tools/packaging/kata-deploy/scripts/kata-deploy.sh

```diff
@@ -199,32 +245,10 @@ function configure_containerd() {
 	# Add default Kata runtime configuration
 	configure_containerd_runtime
 
-	#Currently containerd has an assumption on the location of the shimv2 implementation
-	#Until support is added (see https://github.com/containerd/containerd/issues/3073),
-	#create a link in /usr/local/bin/ to the v2-shim implementation in /opt/kata/bin.
-
-	mkdir -p /usr/local/bin
+	configure_different_shims_base
 
 	for shim in "${shims[@]}"; do
 		configure_containerd_runtime $shim
-
-		local shim_binary="containerd-shim-kata-${shim}-v2"
-		local shim_file="/usr/local/bin/${shim_binary}"
-		local shim_backup="/usr/local/bin/${shim_binary}.bak"
-
-		if [ -f "${shim_file}" ]; then
-			echo "warning: ${shim_binary} already exists" >&2
-			if [ ! -f "${shim_backup}" ]; then
-				mv "${shim_file}" "${shim_backup}"
-			else
-				rm "${shim_file}"
-			fi
-		fi
-       cat << EOT | tee "$shim_file"
-#!/bin/bash
-KATA_CONF_FILE=/opt/kata/share/defaults/kata-containers/configuration-${shim}.toml /opt/kata/bin/containerd-shim-kata-v2 \$@
```

Comment:
In the crio configuration, is there a way to specify the configuration file used for a given runtime handler? If not, this would be nice to add.

This is the case in containerd now, which allows us to simplify things a bit. We'd no longer need to create this bash indirection, nor would we need an equivalent symlink to have things like containerd-shim-kata-clh-v2: 
Example config snippet for containerd:

```
    [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata-clh]
          runtime_type = "io.containerd.kata.v2"
          privileged_without_host_devices = true
          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata-clh.options]
            ConfigPath = "/etc/kata-containers/configuration-clh.toml
```

Reply:
CRI-O side doesn't allow that, but I'll open an issue on CRI-O and follow up with the implementation there.
Thanks for the tip!

Reply:
This is now tracked on CRI-O side: https://github.com/cri-o/cri-o/issues/4591